### PR TITLE
Loadout Modsuit Configuration

### PR DESCRIPTION
**Pull request by Geevies**
* The PMCG modsuit can now be configured inside the loadout.

image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/client/preference_setup/loadout/items/factions.dm [hunk 1]
```diff
@@ -351,13 +351,62 @@
...
 	path = /obj/item/clothing/under/pmc_modsuit
 	slot = slot_w_uniform
 	faction = "Private Military Contracting Group"
 
+/datum/gear/faction/pmc_modsuit/New()
+	..()
+	gear_tweaks += list(gear_tweak_modsuit_configuration)
+
+
+var/datum/gear_tweak/modsuit_configuration/gear_tweak_modsuit_configuration = new()
```
Comment (FluffyGhoster) [inline code/modules/client/preference_setup/loadout/items/factions.dm:367]: Use glob

File: code/modules/client/preference_setup/loadout/items/factions.dm [hunk 2]
```diff
@@ -351,13 +351,62 @@
...
+
+/datum/gear_tweak/modsuit_configuration
+	var/list/configuration_options = list()
```
Comment (FluffyGhoster) [inline code/modules/client/preference_setup/loadout/items/factions.dm:370]: DMDoc and describe the format of this list

File: code/modules/client/preference_setup/loadout/items/factions.dm [hunk 3]
```diff
@@ -351,13 +351,62 @@
...
+
+/datum/gear_tweak/modsuit_configuration/New()
+	..()
+
+	var/obj/item/clothing/under/pmc_modsuit/initial_uniform = new()
```
Comment (FluffyGhoster) [inline code/modules/client/preference_setup/loadout/items/factions.dm:375]: I don't think we need to instantiate this? Initial / :: should be sufficient

File: code/modules/client/preference_setup/loadout/items/factions.dm [hunk 4]
```diff
@@ -351,13 +351,62 @@
...
+
+	var/index = 0
+	for(var/name in initial_uniform.names)
+		configuration_options[name] = index
+		index++
+
+	qdel(initial_uniform)
```
Comment (FluffyGhoster) [inline code/modules/client/preference_setup/loadout/items/factions.dm:382]: Likewise if we don't need to instantiate it we don't need to delete it

File: code/modules/client/preference_setup/loadout/items/factions.dm [hunk 5]
```diff
@@ -351,13 +351,62 @@
...
+
+/datum/gear_tweak/modsuit_configuration/get_contents(var/metadata)
+	return "Modsuit Configuration: [metadata]"
+
+/datum/gear_tweak/modsuit_configuration/get_default()
+	return configuration_options[1]
+
+/datum/gear_tweak/modsuit_configuration/get_random()
+	return configuration_options[1]
```
Comment (FluffyGhoster) [inline code/modules/client/preference_setup/loadout/items/factions.dm:391]: This isn't random, violating the sense of the proc

File: code/modules/clothing/factions/pmcg.dm [hunk 6]
```diff
@@ -50,7 +50,8 @@
 		item_state = initial(item_state) + "_[modsuit_mode]"
 
 	update_clothing_icon()
-	user.update_action_buttons()
+	if(user)
+		user.update_action_buttons()
```
Comment (FluffyGhoster) [inline code/modules/clothing/factions/pmcg.dm:54]:
```suggestion
	user?.update_action_buttons()
```

File: code/modules/client/preference_setup/loadout/items/factions.dm (hunk 1 above)
Comment (Geevies) [reply code/modules/client/preference_setup/loadout/items/factions.dm:368]: breaks standards and updating all of them is out of scope

image

File: code/modules/client/preference_setup/loadout/items/factions.dm (hunk 4 above)
Comment (Geevies) [reply code/modules/client/preference_setup/loadout/items/factions.dm:384]: iirc, initial doesn't work on lists, it has to be initialized and deleted

File: code/modules/client/preference_setup/loadout/items/factions.dm (hunk 3 above)
Comment (Geevies) [reply code/modules/client/preference_setup/loadout/items/factions.dm:377]: iirc, initial doesn't work on lists, it has to be initialized and deleted

File: code/modules/clothing/factions/pmcg.dm (hunk 6 above)
Comment (FluffyGhoster) [reply code/modules/clothing/factions/pmcg.dm:57]: immagine

**Review by Arrow768 — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**